tiledb/sm/c_api/tiledb.h

```diff
@@ -4585,6 +4642,35 @@ TILEDB_EXPORT int32_t tiledb_query_get_fragment_timestamp_range(
     uint64_t* t1,
     uint64_t* t2);
 
+/**
+ * return a TileDB subarray object from the given query.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_t* subarray;
+ * tiledb_query_get_subarray(array, &subarray);
+ * @endcode
+ *
+ * @param ctx The TileDB context.
+ * @param array An open array object.
+ * @param subarray The retrieved subarray object if available.
+ * @return `TILEDB_OK` for success or `TILEDB_OOM` or `TILEDB_ERR` for error.
+ *
+ * @note This leaves the returned subarray with references to everything
+ * that was an active part of the query's contained subarray,
+ * currently this consists of a raw pointer to an internal core entity,
+ * (const tiledb::sm::Array* array_; )
+ * so the receiving client should keep those alive as long as the
+ * returned subarray is alive, possibly most easily tracked by keeping
+ * the source query and its dependencies alive.
+ */
+TILEDB_EXPORT
+int32_t tiledb_query_get_subarray(
```

Comment:
I can see adding the _t for consistency if desired, 
however I think planning to remove them in the future is a bad idea as it will just be introducing an unnecessary breaking behavior (failed compilations on tiledb upgrades) without any functional gains to clients of the APIs.